internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -119,64 +120,111 @@ func (ws *workflows) getSessionData(ctx workflow.Context, sessionID sdktypes.Ses
 	return data, nil
 }
 
-func (ws *workflows) cleanupSession(ctx workflow.Context, data *sessiondata.Data) {
+func (ws *workflows) cleanupSession(data *sessiondata.Data) {
 	z := ws.z.With(zap.String("session_id", data.SessionID.String()))
 
 	if depID := data.Session.DeploymentID(); depID.IsValid() {
-		workflow.Go(ctx, func(ctx workflow.Context) {
-			// TODO: can this be done async? do we care?
-			if err := workflow.ExecuteLocalActivity(
-				ctx,
-				ws.deactivateDrainedDeployment,
-				depID,
-			).Get(ctx, nil); err != nil {
+		// We cannot rely on workflow context here as it might have been cancelled.
```

Comment:
Nit: I see you already fixed the code (cancelled --> canceled), please fix this comment too, and also in line 142 below.

Reply:
done